addons/website/static/tests/tours/snippet_social_media.js

```diff
@@ -3,6 +3,20 @@
 import tour from 'web_tour.tour';
 import wTourUtils from 'website.tour_utils';
 
+// The following step is preventing a race condition to happen.
+// It might be removed once the real issue (at the right panel framework level)
+// is fixed later.
+// When the tour was clicking on the toggle to hide facebook in the next step,
+// it would actually "ignore" the result of the click on the toggle and would
+// just consider the action of focusing out the input.
+const preventRaceConditionStep = {
+    content: "Wait a second to avoid race condition",
+    trigger: 'iframe .s_social_media',
+    async run() {
+        await new Promise(r => setTimeout(r, 1000));
+    }
+};
```

Comment:
This does nothing: `run` functions' returns are just ignored so here you don't delay anything. If this solves a race condition, I guess it is just by chance since you add a tour step which may give a couple extra ms for the next step to occur... but definitely not a second :wink:

Reply:
Indeed, actually just a `0` timeout is enough to fix all those bugs.
Probably something related to the JS single thread execution on which `setTimeout` acts special. It's a bit above my head to figure exactly.
Seems reasonable as is now? Multi build & local `watch=True` are all green :+1:

Reply:
I don't see how a setTimeout 0 that way would change anything about the execution :grimacing: Are you sure that leaving the run method empty does not do exactly the same?

Reply:
I pushed another way of bypassing the race condition, more verbose but at least straightforward.
Ran 2x multi builds in master and all [green](https://runbot.odoo.com/runbot/batch/983078/build/24633281) still